File: Ruby/lib/Monster.rb
#encoding: utf-8

# Autores:
# 
#     Guillermo Montes Martos
#     Manuel Ruiz López
#


require_relative 'Prize.rb'
require_relative 'BadConsequence.rb'

module NapakalakiGame
  class Monster

    attr_reader :name
    attr_reader :combatLevel
    attr_reader :badConsequence
    
    private_class_method:new
    def initialize(n, l, b, p, lC)
      @name = n
      @combatLevel = l
      @badConsequence = b
      @prize = p
      @levelChangeAgainstCultistPlayer = lC
    end
    
    def self.newMonster(n, l, b, p)
      new(n, l, b, p, 0)
    end
    
    def self.newCultistMonster(n, l, b, p, lC)
      new(n, l, b, p, lC)
    end
    
    def getLevelsGained()
      @prize.levels
    end
    
    def getTreasuresGained()
      @prize.treasures
    end
    
    def getCombatLevelAgainstCultistPlayer()
      @combatLevel + @levelChangeAgainstCultistPlayer
    end
    
    def to_s()
      @name + " (#{@combatLevel}) \n\n\tMal rollo: #{@badConsequence} \n\tBuen rollo: #{@prize}\n\n"
    end
    
  end
end

File: Java/src/NapakalakiGame/Monster.java
/*
 * Autores:
 * Guillermo Montes Martos
 * Manuel Ruiz López
 *
 */
package NapakalakiGame;

public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence badConsequence;
    private int levelChangeAgainstCultistPlayer;
    
    // Constructor 
    
    public Monster(String n, int l, BadConsequence b, Prize p) {
        this.name = n;
        this.combatLevel = l;
        this.badConsequence = b;
        this.prize = p;
        this.levelChangeAgainstCultistPlayer = 0;
    }
    
    public Monster(String n, int l, BadConsequence b, Prize p, int lC) {
        this.name = n;
        this.combatLevel = l;
        this.badConsequence = b;
        this.prize = p;
        this.levelChangeAgainstCultistPlayer = lC;
    }
    
    
    // Métodos get
    
    public String getName() {
        return name;
    }
    
    public int getCombatLevel() {
        return combatLevel;
    }
    
    public BadConsequence getBadConsequence() {
        return badConsequence;
    }
    
    public Prize getPrize() {
        return prize;
    }
    
    public int getLevelsGained() {
        return prize.getLevels();
    }
    
    public int getTreasuresGained() {
        return prize.getTreasures();
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return levelChangeAgainstCultistPlayer + combatLevel;
    }
    
    @Override
    public String toString() {
        return (name+" (Nivel de combate: "+combatLevel+")\nMal rollo: "+badConsequence.toString()+"\nBuen rollo: "+prize.toString());
    }
    
    // Practica 1
    /*
    public boolean perdidasoloNiveles () {
        return badConsequence.perdidasoloNiveles();
    }
    
    public boolean gananciaNivelesSup1 () {
        return (prize.getLevel() > 1);
    }
    
    public boolean perdidatipoTesoros () {
        return badConsequence.perdidatipoTesoros();
    }    
    
    // Método de salida string
    
    public String toString() {
        return ("Name = " + name + ". Level = " + Integer.toString(combatLevel) 
                + "\n" + "Prize: " + prize.toString() + "\n" 
                + "Bad Consequence: " + badConsequence.toString() + "\n");
        
    }
    */

}
